pkg/httphelper/client.go

```diff
@@ -456,3 +457,50 @@ func callNodeMgmtEndpoint(client *NodeMgmtClient, request nodeMgmtRequest, conte
 
 	return body, nil
 }
+
+func (client *NodeMgmtClient) CallIsFullQueryLogEnabledEndpoint(pod *corev1.Pod) (bool, error) {
+	client.Log.Info("client::callIsFullQueryLogEnabledEndpoint")
+	podHost, err := BuildPodHostFromPod(pod)
+	if err != nil {
+		return false, err
+	}
+	request := nodeMgmtRequest{
+		endpoint: "/api/v0/ops/node/fullquerylogging",
+		host:     podHost,
+		method:   http.MethodGet,
+		timeout:  time.Minute * 2,
+	}
+	isEnabledBytes, err := callNodeMgmtEndpoint(client, request, "")
+	if err != nil {
+		client.Log.Error(err, "failed to call endpoint /api/v0/ops/node/fullquerylogging")
+		return false, err
+	}
+	var isEnabledMap map[string]interface{}
+	err = json.Unmarshal([]byte(isEnabledBytes), &isEnabledMap)
+	if err != nil {
+		client.Log.Error(err, "failed to unmarshall JSON response from /api/v0/ops/node/fullquerylogging", "response", string(isEnabledBytes))
+		return false, err
+	}
+	isEnabledBool := isEnabledMap["entity"]
```

Comment:
This and the last line are unsafe assumptions. When fetching a key, you should check if that key actually exists in the payload. And when doing a cast, you need to that the cast is possible - otherwise you will panic the runtime and the cass-operator will go crashing down.

Reply:
Oh jeez. So I couldn't figure this out because for some horrible reason a cast like this: `isEnabledBool.(bool)` returns `(bool, bool)` not `bool, err` like every other function call in Go. 

Same goes for when a map indexing operation can't find a key it appears. Do note that when indexing the map I actually was doing this check:

```
	if isEnabledBool == nil {
		err := errors.New("failed to access entity key in response from /api/v0/ops/node/fullquerylogging")
		return false, err
	}
```

I'm pretty sure that would avoid a panic but I'm curious if you think otherwise.

Either way, still a bit more practice with Go required for me it appears...